snoop/data/filesystem.py

```diff
@@ -68,37 +68,38 @@ def file_to_blob(directory, name):
     return file
 
 
-@shaorma
+@shaorma('filesystem.handle_file')
 def handle_file(file_pk):
     file = models.File.objects.get(pk=file_pk)
     depends_on = {}
 
     if archives.is_archive(file.blob.mime_type):
-        unarchive_task = archives.unarchive.laterz(file.blob.pk)
+        digest_blob = file.blob
+        unarchive_task = archives.unarchive.laterz(file.blob)
         create_archive_files.laterz(
             file.pk,
             depends_on={'archive_listing': unarchive_task},
         )
 
     elif file.blob.mime_type == 'message/x-emlx':
         digest_blob = emlx.reconstruct(file)
-        depends_on['email_parse'] = email.parse.laterz(digest_blob.pk)
+        depends_on['email_parse'] = email.parse.laterz(digest_blob)
 
     elif file.blob.mime_type == 'message/rfc822':
         digest_blob = file.blob
-        depends_on['email_parse'] = email.parse.laterz(digest_blob.pk)
+        depends_on['email_parse'] = email.parse.laterz(digest_blob)
 
     elif tika.can_process(file.blob):
         digest_blob = file.blob
-        depends_on['tika_rmeta'] = tika.rmeta.laterz(digest_blob.pk)
+        depends_on['tika_rmeta'] = tika.rmeta.laterz(digest_blob)
 
     else:
         digest_blob = file.blob
 
-    digest.laterz(file.collection.pk, digest_blob.pk, depends_on=depends_on)
+    digest.laterz(digest_blob, file.collection.pk, depends_on=depends_on)
 
 
-@shaorma
+@shaorma('filesystem.create_archive_files')
 def create_archive_files(file_pk, archive_listing):
```

Comment:
`archive_listing` is a blob, reverse the arguments to save it to the task

Reply:
Turns out `archive_listing` comes from `depends_on` so it's already a blob foreign key from the `TaskDependency` model.